Sources/SupplementaryViewInfo.swift

```diff
@@ -17,6 +17,12 @@
 import Foundation
 import UIKit
 
+public protocol ReusableSupplementaryViewProtocol {
+
+    /// The registration info for the supplementary view.
+    var viewInfo: SupplementaryViewInfo? { get } // TODO: make this not optional
```

Comment:
Are you avoiding addressing this TODO now to keep this PR reasonably sized?

Reply:
correct. will follow-up